addons/account_edi_ubl_cii/data/cii_22_templates.xml

```diff
@@ -93,6 +93,17 @@
                     <ram:ID t-att-schemeID="str('0002')"
                             t-esc="specified_legal_organization_val"/>
                 </ram:SpecifiedLegalOrganization>
+
+                <ram:DefinedTradeContact>
+                    <ram:PersonName t-esc="partner.name"/>
+                    <ram:TelephoneUniversalCommunication t-if="partner.phone or partner.mobile">
+                        <ram:CompleteNumber t-esc="partner.phone or partner.mobile"/>
+                    </ram:TelephoneUniversalCommunication>
+                    <ram:EmailURIUniversalCommunication t-if="partner.email">
+                        <ram:URIID schemeID='SMTP' t-esc="partner.email"/>
+                    </ram:EmailURIUniversalCommunication>
+                </ram:DefinedTradeContact>
```

Comment:
Dug a bit into it, and here is what I can conclude:
Nearly good, but this block is only supposed to be there in `SellerTradeParty` and `BuyerTradeParty` for xrechnung.
For facturx it seems it can be in the three places where this template is called so no issue there.

So for the DE version, when rendering this template from `ShipToTradeParty`, you should not add DefinedTradeContact

Note from juvr: if you uncomment this line `#response = self.env['account.edi.common']._check_xml_ecosio(invoice, xml_content, builder._export_invoice_ecosio_schematrons())` it will test xrechnung automatically when you post the invoice. It should be all green after omitting `DefinedTradeContact` from `ShipToTradeParty` when building the XML for xrechnung.

Reply:
Any idea how I can get rid of these errors with xrechnung ?
![xrechnungerrors](https://user-images.githubusercontent.com/17316059/187410372-423dfb83-d341-489f-a5b9-b30177e09b04.png)

Reply:
Oups, for "Payment instructions" I think it's more about the company: set a DE IBAN on the partner linked to your German company, then make sure that on the invoice, the field `Recipient bank` is filled with that account (automatic if you do a new one, or you need to set manually if the invoice already exists. It should find the bank account on your company's partner)